arroyo-state/src/committing_state.rs

```diff
@@ -25,13 +33,35 @@ impl CommittingState {
     pub fn done(&self) -> bool {
         self.subtasks_to_commit.is_empty()
     }
-}
 
-impl From<(i64, HashSet<(String, u32)>)> for CommittingState {
-    fn from((checkpoint_id, subtasks_to_commit): (i64, HashSet<(String, u32)>)) -> Self {
-        Self {
-            checkpoint_id,
-            subtasks_to_commit,
-        }
+    pub fn committing_data(&self) -> HashMap<String, OperatorCommitData> {
+        let operators_to_commit: HashSet<_> = self
+            .subtasks_to_commit
+            .iter()
+            .map(|(operator_id, _subtask_id)| operator_id.clone())
+            .collect();
```

Comment:
why do we collect this then immediately turn it back into an iterator?

Reply:
It's deduplicating them by a collecting into a hashset.